File: pythonDemos/quad.py
import pygame

class Quad:
    __slots__ = ['_xmid', '_ymid', '_length']
    def __init__(self, var1:float, var3:float, var5:float)->None:
        self._xmid = var1
        self._ymid = var3
        self._length = var5

    def length(self)->float:
        return self._length
    
    def contains(self, x:float, y:float)->bool:
        halfLen = self._length / 2.0
        return( (x <= self._xmid + halfLen) and
                (x > self._xmid - halfLen) and
                (y < self._ymid + halfLen) and
                (y >= self._ymid - halfLen))
    
    def NW(self)->'Quad':
        var1 = self._xmid - self._length / 4.0
        var3 = self._ymid - self._length / 4.0
        var5 = self._length / 2.0
        return Quad(var1, var3, var5)
    
    def NE(self)->'Quad':
        var1 = self._xmid + self._length / 4.0
        var3 = self._ymid - self._length / 4.0
        var5 = self._length / 2.0
        return Quad(var1, var3, var5)
    
    def SW(self)->'Quad':
        var1 = self._xmid - self._length / 4.0
        var3 = self._ymid + self._length / 4.0
        var5 = self._length / 2.0
        return Quad(var1, var3, var5)

    def SE(self)->'Quad':
        var1 = self._xmid + self._length / 4.0
        var3 = self._ymid + self._length / 4.0
        var5 = self._length / 2.0
        return Quad(var1, var3, var5)
    
    def draw(self, screen:pygame.display)->None:
        pygame.draw.rect(screen, (255,255,255), pygame.Rect(self._xmid - self._length / 2.0, self._ymid - self._length / 2.0, self._length, self._length), 1)
    
    def __str__(self)->str:       
        var1 = "\n"
        for var2 in range(int(self._length)):
            for var3 in range(int(self._length)):
                if var2 != 0 and var3 != 0 and var2 != self._length - 1 and var3 != self._length - 1:
                    var1 += " "
                else:
                    var1 += "*"
            var1 += "\n"
        return var1

File: src/Quad.java
import java.awt.Color;
import java.awt.Graphics;

public class Quad{
    private double xmid, ymid,length;
    
    public Quad(double xmid, double ymid, double length){
        this.xmid = xmid;
        this.ymid = ymid;
        this.length = length;
    }
    public double length(){
        return this.length;
    }
    public boolean contains(double x, double y){
        double halfLength = this.length / 2.0;
        return (x <= this.xmid + halfLength) && (x > this.xmid - halfLength) && (y < this.ymid + halfLength) && (y >= this.ymid - halfLength);
    }
    public Quad NW(){
        double _xmid = this.xmid - this.length / 4.0;
        double _ymid = this.ymid - this.length / 4.0;
        double _length = this.length / 2.0;
        return new Quad(_xmid, _ymid, _length);
    }
    public Quad NE(){
        double _xmid = this.xmid + this.length / 4.0;
        double _ymid = this.ymid - this.length / 4.0;
        double _length = this.length / 2.0;
        return new Quad(_xmid, _ymid, _length);
    }
    public Quad SW(){
        double _xmid = this.xmid - this.length / 4.0;
        double _ymid = this.ymid + this.length / 4.0;
        double _length = this.length / 2.0;
        return new Quad(_xmid, _ymid, _length);
    }
    public Quad SE(){
        double _xmid = this.xmid + this.length / 4.0;
        double _ymid = this.ymid + this.length / 4.0;
        double _length = this.length / 2.0;
        return new Quad(_xmid, _ymid, _length);
    }
    public void draw(Graphics g,Color color){
        g.setColor(color);
        g.drawRect((int)(this.xmid - this.length / 2.0), (int)(this.ymid - this.length / 2.0), (int)this.length, (int)this.length);
        
    }
}